api.ts

```diff
@@ -2203,3 +2288,49 @@ export function maybeRewriteConnectionForFormula<
   }
   return formula;
 }
+
+// This helper method finds any inline autocomplete functions in a static sync table schema.
+// These functions will need to be extracted into the "namedAutocompletes" property on the sync
+// table and replaced with strings.
+function listPropertiesWithAutocompleteFunctions(schema: ObjectSchemaDefinition<string, string>): string[] {
+  const result: string[] = [];
+  for (const propertyName of Object.keys(schema.properties)) {
+    const {autocomplete} = schema.properties[propertyName];
+    if (!autocomplete) {
+      continue;
+    }
+    if (typeof autocomplete !== 'function') {
+      continue;
+    }
+
+    result.push(propertyName);
+  }
+  return result;
+}
+
+// Finds any inline autocomplete functions within the inputSchema and replaces them
+// with strings references into the returned namedAutocompletes.
+function replaceInlineAutocompleteFunctionsWithNamedAutocompleteFunctions({
+  inputSchema, // DO NOT MUTATE inputSchema!
+  schema,
+  identityName,
+}: {
+  inputSchema: ObjectSchemaDefinition<any, any>;
+  schema: ObjectSchemaDefinition<any, any>;
+  identityName: string;
+}): SyncTableAutocompleters {
+  // Converting JS functions to strings happens on inputSchema instead of the deep copied version because the
+  // deep copy will have already thrown away any JS functions.
+  const namedAutocompletes: SyncTableAutocompleters = {};
+
+  for (const propertyName of listPropertiesWithAutocompleteFunctions(inputSchema)) {
+    schema.properties[propertyName].autocomplete = propertyName as AutocompleteReference;
+    namedAutocompletes[propertyName] = makePropertyAutocompleteFormula({
+      execute: inputSchema.properties[propertyName].autocomplete as any,
+      schema: normalizeSchema(schema.properties[propertyName]),
+      name: `${identityName}.${propertyName}.Autocomplete`,
```

Comment:
Yes. This particular `name` field isn't used for anything except maybe making it clear what you're looking at in a debugging, since the formula isn't exposed for use anywhere outside the PropertyAutocomplete metadata function.